### PR TITLE
Updated Node Js Dockerfiles to Node 10.15.0 (Alpine)

### LINKED ISSUE #99: Update node and node-armhf templates to Node 10
Node 10 will be released on the [2018-10-30](https://github.com/nodejs/Release) and will become the new LTS. If that happens we should update the base image to Node 10.

## Your Environment

* Are you using Docker Swarm or Kubernetes (FaaS-netes)?

* Operating System and version (e.g. Linux, Windows, MacOS):

* Link to your project or a code example to reproduce issue:

### PR DESCRIPTION
**Pull request by Templum**
As described in #99 the new version of NodeJs `10.X.X` does not introduce breaking changes, but instead provides performance gains and bug fixes. 
## Description
During my work I discovered the following things:

* Both arm Templates don't use an user, but instead run with the root user. Is this inteded, otherwise I can also fix that

* `node-armhf` installed node and npm for some reason switched over to use `arm32v6/node:10.15.0-alpine

* `node-arm64` was not based on alpine, like the others, switched over to `arm64v8/node:10.15.0-alpine`

* `node-armhf` & `node-arm64` differed much from the other. So I adjusted the code so it is similiar

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Which issue(s) this PR fixes 

Fixes #99 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: template/node-armhf/Dockerfile [hunk 1]
```diff
@@ -1,30 +1,36 @@
-FROM arm32v6/alpine:3.6
+FROM arm32v6/node:10.15.0-alpine
```
Comment (alexellis) [inline template/node-armhf/Dockerfile]: Does this work without the prefix? We may be able to use the multi arch images? Ask in thy ARM channel for someone to validate if you don't own an RPi.

Comment (rgee0) [reply template/node-armhf/Dockerfile]:
```
$ echo 'node:10.15.0-alpine' | faas invoke mquery
Manifest List: Yes
Supported platforms:
 - linux/amd64
 - linux/arm/v6
 - linux/arm64
 - linux/386
 - linux/ppc64le
 - linux/s390x
```

Comment (alexellis) [reply template/node-armhf/Dockerfile]: Seems like we can remove the prefixes?

File: template/node-arm64/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,10 @@
-FROM arm64v8/node:8.9.1
+FROM arm64v8/node:10.15.0-alpine
```
[the remark this answers is not in the document]
Comment (Templum) [reply template/node-arm64/Dockerfile]: I will change this. And what about the open issue with the root user ? (Aka that arm based images don't use an app user)

[the remark this answers is not in the document]
Comment (alexellis) [reply template/node-arm64/Dockerfile]: I think they should all be non root and matching. The only change I can think of is the watchdog binary name.

[the remark this answers is not in the document]
Comment (Templum) [reply template/node-arm64/Dockerfile]: Alright then I will fix this as well. Today I'm on Ng girls event as an mentor. I believe in the evening I should find time make the suggested changes

[the remark this answers is not in the document]
Comment (Templum) [reply template/node-arm64/Dockerfile]: @alexellis in 4f4525b switched over to the non-prefixed versions and further aligned all images to the "node" templates.

**Comment by alexellis** [comment]
I want to keep the changes as atomic as possible, because immediately after merge all users are impacted.

File: template/node-arm64/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexellis) [reply template/node-arm64/Dockerfile]: 👍 thanks for explaining, this is fine for the arm64 template, let's mirror the x86_64 one.

**Pull request closed without merging**